runtime/lua/vim/fs.lua

```diff
@@ -415,16 +430,30 @@ function M.root(source, marker)
     error('invalid type for argument "source": expected string or buffer number')
   end
 
-  local paths = M.find(marker, {
-    upward = true,
-    path = vim.fn.fnamemodify(path, ':p:h'),
-  })
+  local marker_t = type(marker)
+  if marker_t == 'string' or marker_t == 'function' then
+    local paths = M.find(marker, {
+      upward = true,
+      path = vim.fn.fnamemodify(path, ':p:h'),
+    })
 
-  if #paths == 0 then
-    return nil
+    if #paths ~= 0 then
+      return vim.fs.dirname(paths[1])
+    end
+  else
+    for _, mark in ipairs(marker) do
+      local paths = M.find(mark, {
+        upward = true,
+        path = vim.fn.fnamemodify(path, ':p:h'),
+      })
+
+      if #paths ~= 0 then
+        return vim.fs.dirname(paths[1])
+      end
+    end
```

Comment:
this `else` assumes that `marker` is a list? anyway, similar to the old code in lsp.lua, we don't two separate paths. if `marker` is not a list, just convert it into a list.

```lua
    local markers = type(marker) == 'table' and marker or { marker }
    for _, mark in ipairs(markers) do
      local paths = M.find(mark, {
        upward = true,
        path = vim.fn.fnamemodify(path, ':p:h'),
      })

      if #paths ~= 0 then
        return vim.fs.dirname(paths[1])
      end

```